File: app/src/main/java/my/com/toru/daggertutorial/di/AppComponent.java
package my.com.toru.daggertutorial.di;

import android.app.Activity;
import android.app.Application;

import javax.inject.Singleton;

import dagger.Component;
import my.com.toru.daggertutorial.remote.RemoteModule;
import my.com.toru.daggertutorial.ui.detail.DetailActivity;
import my.com.toru.daggertutorial.ui.main.MainActivity;
import my.com.toru.daggertutorial.ui.other.OtherActivityComponent;
import my.com.toru.daggertutorial.ui.other.di.DummyDependModule;

@Singleton
@Component(modules = {RemoteModule.class})
public interface AppComponent {
    void inject(Application ac);
    void inject(Activity activity);
    void inject(DetailActivity activity);

    OtherActivityComponent.Builder otherActivityBuilder();
}

File: kdaggertutorial/src/main/java/my/com/toru/kdaggertutorial/di/AppComponent.kt
package my.com.toru.kdaggertutorial.di

import dagger.Component
import dagger.Module
import dagger.Provides
import my.com.toru.kdaggertutorial.app.KDaggerTutorialApp
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {
    fun inject(app:KDaggerTutorialApp)
    fun mainActivityBuilder():MainComponent.Builder
//    fun detailActivityBuilder():DetailComponent.Builder
}

@Module
class AppModule(val app:KDaggerTutorialApp){
    @Provides
    fun provideApp() = app
}